cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -2201,11 +2201,11 @@ void AddSplitWhitespaceAscii(FunctionRegistry* registry) {
   auto func =
       std::make_shared<ScalarFunction>("ascii_split_whitespace", Arity::Unary(),
                                        &ascii_split_whitespace_doc, &default_options);
-  using t32 = SplitWhitespaceAsciiExec<StringType, ListType>;
-  using t64 = SplitWhitespaceAsciiExec<LargeStringType, ListType>;
-  DCHECK_OK(func->AddKernel({utf8()}, {list(utf8())}, t32::Exec, t32::State::Init));
-  DCHECK_OK(
-      func->AddKernel({large_utf8()}, {list(large_utf8())}, t64::Exec, t64::State::Init));
+
+  for (const auto& ty : StringTypes()) {
+    auto exec = GenerateTypeAgnosticVarBinary<SplitWhitespaceAsciiExec, ListType>(ty);
+    DCHECK_OK(func->AddKernel({ty}, {list(ty)}, exec, SplitState::Init));
+  }
```

Comment:
Ditto - I don't think this needs to support binary values.

Reply:
(etc. for kernels below labeled ascii_ or utf8_.)

Reply:
Based on [Zulip discussion](https://ursalabs.zulipchat.com/#narrow/stream/180245-dev/topic/Stringlike.20kernels.20on.20binary.20data), we can support binary types for ASCII-named compute functions.

Reply:
Created [ARROW-14151](https://issues.apache.org/jira/browse/ARROW-14151) for the ASCII functions.